public/app/features/datasources/components/BasicSettings.tsx

```diff
@@ -1,7 +1,23 @@
-import React from 'react';
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
 
+import { GrafanaTheme2 } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
-import { InlineField, InlineSwitch, Input } from '@grafana/ui';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { InlineField, InlineSwitch, Input, Badge, useStyles2 } from '@grafana/ui';
+
+const useInitHasAlertingEnabled = (dataSourceName: string) => {
+  const [hasAlertingEnabled, setHasAlertingEnabled] = useState(false);
+
+  useEffect(() => {
+    const initDataSourceAlertingEnabled = async () => {
```

Comment:
I think it's easier to also return early from this effect so we don't need the function declaration inside the effect

```
if (!dataSourceName) {
  return;
}
```

We can also use `useAsync()` to remove the need for `useEffect` and `useState` and combine both in a single hook.

Reply:
👌

Reply:
This code has been simplified using a non async method.